app/src/main/java/com/example/dogbook/SignupActivity/SignupActivity.java

```diff
@@ -3,15 +3,122 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 
+import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.example.dogbook.LoginActivity.LoginActivity;
+import com.example.dogbook.MainActivity.MainActivity;
 import com.example.dogbook.R;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+import java.util.Date;
 
 public class SignupActivity extends AppCompatActivity {
 
+    private static final String TAG = "SignupActivity";
+
+    EditText etUsername;
+    EditText etPassword;
+    EditText etConfirmPassword;
+    EditText etBreed;
+    DatePicker dpBirthday;
+    CheckBox cbLookingForPlaymates;
+    EditText etOwnerName;
+    EditText etOwnerContact;
+    Button btnSignup;
+    Button btnLogin;
```

Comment:
These can all be private, since they're instance variables

Also, as you're getting more variables, might want to consider pulling this out its own separate view class, that way you decrease the amount of logic that's going on in here

Reply:
Let's try to integrate private variables and a lot of the feedback into how you write code for the first version.  This will help you to actually be more productive since you won't have to go back and spend time updating the code on top of what you already, normally write